### ISSUE TITLE
Sending packetplayoutmapchunk with mask 65535 has strange effects

### ISSUE THREAD

**Issue by Miclebrick** [sp614x/optifine#1431]
When using masks that only include the changed chunk sections, this works fine. However, in optifine c7+, using bitmask 65535 which includes all sections results in invisible blocks which you can see the outline of when pointing cursor at the block until using f3 + a, which fixes it.

(Also, side question, is this the fastest way to update many blocks? Multiblock change seems to be much slower, but this way occasionally causes freezing (?))

**Comment by sp614x** [comment]
What masks, where, how are they used?
Is this a mod, resource pack or something else?

**Comment by Dushess** [comment]
Probably worldrdit...

**Comment by Miclebrick** [comment]
Worldedit and custom plugins. The mask is in the packet. I think this is the packet: http://wiki.vg/Protocol#Chunk_Data

**Comment by Miclebrick** [comment]
Calling work.refreshChunk() server side or playerchunk.sendChunk() with the default mask has the issue, but using the packet with a custom mask doesn't

**Comment by sp614x** [comment]
Is it working correctly in C6?

**Comment by sp614x** [comment]
The issue started as #1112.

**Comment by Miclebrick** [comment]
Yes

**Comment by Miclebrick** [comment]
It's almost definitely that issue, but I thought I'd make a new more specific issue just in case.

**Comment by Miclebrick** [comment]
New developments: I think it's not actually the bitmask alone causing that, but this flag: http://wiki.vg/Chunk_Format#Ground-up_continuous

Internally, MC sets this to true if the bitmask includes the whole chunk: 
`        this.f = i == 65535;
`

**Comment by sp614x** [comment]
There are warnings about using the flag.
Is the server unloading the chunk before sending the new one with the flag?

**Comment by Miclebrick** [comment]
It DOES work without Optifine, even if MC is breaking its own protocol or something... I'll try experimenting with sending unload packets first soon, though

**Comment by Miclebrick** [comment]
It works if you send the unload packet. However, there seems to be strange issues if the player is in the chunk, i.e. severe jittering like when chunk loads aren't keeping up and you are entering an unloaded chunk.

**Comment by sp614x** [comment]
The flag "Ground-up continuous" is for initial chunk loading ONLY. 
The wiki explicitely warns not to use it for chunk updates as it will leak memory and the client will crash eventually.
http://wiki.vg/Chunk_Format#Ground-up_continuous

**Comment by Miclebrick** [comment]
True, but it does work in vanilla and optifine prior to c7, so shouldn't it be supported anyway?

**Comment by sp614x** [comment]
According to the wiki it doesn't work in vanilla because it leaks memory and the client is going to crash eventually. 
I will try to find why it doesn't work with OF and to fix it.
